.github/workflows/core-plugins-build.yml

```diff
@@ -0,0 +1,187 @@
+on:
+  workflow_dispatch:
+    inputs:
+      plugin_id:
+        description: "ID of the plugin you want to publish"
+        required: true
+        type: choice
+        options:
+          - grafana-testdata-datasource
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}-${{ inputs.plugin_id }}
+  cancel-in-progress: true
+
+env:
+    NODEJS_VERSION: 16.19.1
+    GO_VERSION: 1.21.0
+    GO_LINT_VERSION: 1.51.1
+    GRABPL_VERSION: 3.0.41
+    GCP_BUCKET: integration-artifacts
+
+jobs:
+    build-and-publish:
+      name: Build and publish a core plugin
+      runs-on:
+        labels: ubuntu-latest-8-cores
+      steps:
+        - name: checkout
+          uses: actions/checkout@v3
+        - name: 'Authenticate to Google Cloud'
+          uses: 'google-github-actions/auth@v1'
+          with:
+            credentials_json: '${{ secrets.PLUGINS_GOOGLE_CREDENTIALS }}'
+        - name: 'Set up Cloud SDK'
+          uses: 'google-github-actions/setup-gcloud@v1'
+        - name: Verify plugin_id
+          run: |
+            echo "Verifying plugin_id: ${{ inputs.plugin_id }}"
+        - name: Setup nodejs environment
+          uses: actions/setup-node@v3
+          with:
+            node-version: ${{ env.NODEJS_VERSION }}
+            cache: yarn
+        - name: Install frontend dependencies
+          shell: bash
+          working-directory: ./public/plugins/${{ inputs.plugin_id }}
+          run: |
+           yarn install --immutable
+        - name: Setup golang environment
+          uses: actions/setup-go@v3
+          with:
+            go-version: ${{ env.GO_VERSION }}
+        - name: Install Mage
+          shell: bash
+          run: |
+            go install github.com/magefile/mage
+        - name: Install golangci-lint
+          shell: bash
+          run: |
+            curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v${{ env.GO_LINT_VERSION }}
+        - name: Download grabpl executable
+          shell: sh
+          working-directory: ./public/plugins/${{ inputs.plugin_id }}
+          run: |
+            [ ! -d ./bin ] && mkdir -pv ./bin || true
+            curl -fL -o ./bin/grabpl https://grafana-downloads.storage.googleapis.com/grafana-build-pipeline/v${{ env.GRABPL_VERSION }}/grabpl
+            chmod 0755 ./bin/grabpl
+        - name: Validate environment
```

Comment:
Since this is not validating, better call it "print environment info". That way in the future doesn't create a fake feeling of things being validated.

Reply:
it's validating the tools that were installed before but maybe the title is misleading